web3/contract.py

```diff
@@ -722,6 +776,161 @@ def __prepared_function(self, **kwargs):
         return getattr(contract_modifier_func(modifier_dict), self.__function)
 
 
+class ContractMethod(object):
+    """
+    """
+    contract = None
+    method_name = None
+    web3 = None
+    transaction = None
+
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+    def __call__(self, *args, **kwargs):
+        if args:
+            self.args = args
+        if kwargs:
+            self.kwargs = kwargs
+        return self.call()
```

Comment:
As I mentioned in the tests, I think this `__call__` should be dropped altogether.